app/presenters/outcome_presenter.rb

```diff
@@ -74,4 +73,19 @@ def render_data_partial(partial, variable_name)
     partial_path = ::SmartAnswer::FlowRegistry.instance.load_path.join("data_partials", "_#{partial}")
     ApplicationController.new.render_to_string(file: partial_path.to_s, layout: false, locals: {variable_name.to_sym => data})
   end
+
+  def render_erb_template
+    unless @rendered_erb_template
```

Comment:
Is there any chance that `@state` could change between calling `#title` and `#body`? If so, there would be a possibility of rendering the template prematurely thus making it really hard to debug an issue. Could be worth raising an error if `@state` has changed